statefun-sdk-js/src/message.ts

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+import * as types from "./types";
+import {validateTypeName, Address, isEmptyOrNull, Type} from "./core";
+
+class Message {
```

Comment:
I don't know if this makes sense logically, but if it does, there's some more TS magic we could use if we make this class generic:

```
class Message<T = unknown> {
    // …

    // Note the "this is Message<T>"
    is<T>(thatType: Type<T>): this is Message<T> {
        return that.typename === this.valueTypeName;
    }
}
```